### ISSUE TITLE
HPX won't compile - error building resource_partitioner

### ISSUE THREAD

**Issue by khuck** [STEllAR-GROUP/hpx#4220]
## Expected Behavior

... Please describe the behavior you would have expected.

## Actual Behavior

The build is failing when compiling the resource_partitioner:
http://omega.nic.uoregon.edu:8020/#/builders/5/builds/38/steps/7/logs/stdio

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1.  `cmake -DCMAKE_CXX_COMPILER=/packages/gcc/7.1/bin/g++ -DCMAKE_C_COMPILER=/packages/gcc/7.1/bin/gcc -DCMAKE_BUILD_TYPE=Debug -DBOOST_ROOT=/usr/local/packages/boost/1.65.0-gcc7 -DTCMALLOC_ROOT=/usr/local/packages/gperftools/git-gcc7 -DHPX_WITH_MALLOC=tcmalloc -DHWLOC_ROOT= -DCMAKE_INSTALL_PREFIX=. -DHPX_WITH_THREAD_IDLE_RATES=ON -DHPX_WITH_PARCELPORT_MPI=ON -DHPX_WITH_PARCEL_COALESCING=OFF -DHPX_WITH_TOOLS=OFF -DHPX_WITH_TESTS=ON -DHPX_WITH_EXAMPLES=OFF -DHPX_WITH_THREAD_MANAGER_IDLE_BACKOFF=OFF -DHPX_WITH_APEX=TRUE -DAPEX_WITH_ACTIVEHARMONY=TRUE -DACTIVEHARMONY_ROOT=/usr/local/packages/activeharmony/4.6.0-gcc -DAPEX_WITH_OTF2=TRUE -DOTF2_ROOT=/usr/local/packages/otf2-2.1 -DAPEX_WITH_PAPI=TRUE -DPAPI_ROOT=/usr/local/packages/papi/papi-knl/5.5.0/ -DHPX_WITH_APEX_NO_UPDATE=FALSE -DHPX_WITH_APEX_TAG=develop -DHPX_WITH_MAX_CPU_COUNT=72 /var/lib/buildbot/slaves/phylanx/x86_64-gcc7-debug/build/tools/buildbot/src/hpx`

  1. make
  1.

## Specifications

... Please describe your environment

  - HPX Version: current master
  - Platform (compiler, OS):  x86_64 linux, gcc 7.3

**Comment by khuck** [comment]
this might be related to the empty -DHWLOC_ROOT flag.  Checking to see if that's the problem...

**Comment by msimberg** [comment]
Hmm, that looks awfully close to what #4190 fixed, except it was meant to work with GCC 7. Maybe we need to make enable it only form GCC 8 onwards instead...

**Comment by khuck** [comment]
@msimberg is there something I can change to check that?  It's not the HWLOC flag, by the way.

**Comment by khuck** [comment]
@msimberg changing that version to 80000 worked, btw.  Should I do a pull request, or do you want to look further into it?

**Comment by khuck** [comment]
@msimberg interestingly, it worked on Intel KNL with GCC 7.3...but without MPI enabled.  See http://omega.nic.uoregon.edu:8020/#/builders/7/builds/39

**Comment by msimberg** [comment]
It could be GCC 7.1. This is our rostam build with GCC 7.3: http://rostam.cct.lsu.edu/builders/hpx_gcc_7_boost_1_65_centos_x86_64_debug/builds/472/steps/configure/logs/stdio which is happy. You do have a lot of additional options, but the build fails earlier so the GCC version is a good candidate.

**Comment by khuck** [comment]
@msimberg you are right, I was wrong - I was building with 7.1, and that failed.  With the update to 7.3, it worked and changing to 70300 worked.  I'll submit a PR.
